packages/components/src/internal/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -357,14 +357,14 @@ export class SampleTypeDesignerImpl extends React.PureComponent<Props & Injected
         });
     };
 
-    onFinish = (auditUserComment?: string): void => {
+    onFinish = (comment?: string): void => {
         const { defaultSampleFieldConfig, setSubmitting, metricUnitProps } = this.props;
-        const { model, uniqueIdsConfirmed } = this.state;
+        const { model, uniqueIdsConfirmed, auditUserComment } = this.state;
```

Comment:
Why prefer `state.auditUserComment` over passed in `comment`? The following scenario is not working: 
- provide a name that would generate warning, fill in Reason for update, click Finish
- on the warning modal, click Cancel instead of Save anyway.
- click Finish again
- on the warning modal, click Save anyway. Note that the comment is not saved.